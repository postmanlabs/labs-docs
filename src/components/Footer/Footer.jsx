import React, {useState} from 'react';
import { v4 as uuidv4 } from 'uuid';
import styled from 'styled-components';
// import footerDataLocal from '../../../build/footerDev.json';
import footerData from '../../../bff-data/footer.json';



const FooterWrapper = styled.footer`
  border-top: 1px solid ${(props) => props.theme.colors.grey_30};
  background-color: ${(props) => props.theme.colors.grey_00};
  font-size: 14px;
  color: ${(props) => props.theme.colors.grey_50};
.copyright {
  font-size: 12px;
}
.column {
  margin-left: 0px;
}
.footer-col-title {
  font-size: 16px !important;
  font-family: Inter, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans", Helvetica, Arial, sans-serif;
  font-weight: 600;
  line-height: 1.4;
  margin-bottom: 8px; 
}
`

const ColumnWrapper = styled.ul`
  li {
    list-style-type: none;
    margin-bottom: 0;
  }
  
a {
  color: ${(props) => props.theme.colors.grey_50};
  text-transform: none;
  font-weight: 400;
  display: block;
  line-height: 1.7;
  padding: 5px 0;
  &:hover {
    color: ${(props) => props.theme.colors.blue_50};
    text-decoration: none;
  }
  span {
    border: 1px solid ${(props) => props.theme.colors.blue_10};
    border-radius: ${(props) => props.theme.borderRadius.medium};
    padding: 0 7px 0 4px;
    display: inline-block;
    white-space: pre;
    margin-left: 6px;
  }
}
`

const FooterImgWrapper = styled.div`
  @media (min-width: 768px) {
    filter: grayscale(100%);
    transition: ${(props) => props.theme.transitions.all};
    &:hover {
      filter: grayscale(0);
      transition: all 0.2s ease-in-out;
    }
  }
`
const SocialSVGWrapper = styled.div`
  display: inline-block;
  margin-right: 8px;
  width: 16px;
  height: 16px;
  & svg {
    max-height: 100%;
    width: auto;
  }
`

const triggerGA = (category, label) => (
  category
  && label
  && window.pm
  && window.pm.ga('send', 'event', category, 'Click', label)
);

// Helper function for rel attribute in link or button
function relStringGenerator(target) {
  if (target === '') {
    return null;
  }
  if (target === 'blank') {
    return 'noopener';
  }
  return null;
}
// // Helper function for target attribute in link or button
function targetStringGenerator(target) {
  if (target === '') {
    return null;
  }
  if (target === 'blank') {
    return '_blank';
  }
  return null;
}

const Footer = () => {

  const [data] = useState(footerData)

  const columns = data.items.slice(0, 5);
   
  return (
    <FooterWrapper>
      <section id="Footer" className="section">
        <div className="container">
          <div className="row">
            <div className="col-sm-8 offset-sm-2 col-md-12 offset-md-0">
              <div className="row">
                {/* Copyright */}
                <FooterImgWrapper className="col-8 offset-2 col-md-3 offset-md-0 col-lg-2 order-12 order-md-0 pad-md-right align-self-center">
                  <img className="footer-img mb-5" src='https://voyager.postman.com/illustration/postman-footer-rocket-launch.svg' alt="Postman" />
                  <span className="col-12 d-none d-md-block copyright">
                    {data.copyright}
                  </span>
                </FooterImgWrapper>
                {/* Product */}
                <div className="col-6 col-md-2 offset-md-1 col-lg-2 offset-lg-2 order-1 order-md-2 mb-5">
                  {columns.slice(0, 1).map((item) => (
                    <nav aria-labelledby={item.arialabelledby} key={uuidv4()}>
                      <h2 className="footer-col-title" id={item.arialabelledby}>
                        {item.title}
                      </h2>
                      <ColumnWrapper className="column">
                        {(item.items
                          && item.items.map((link) => (
                            <li className="column-row" key={uuidv4()}>
                              <a
                                className="column-link"
                                id={link.id}
                                href={link.url}
                                rel={relStringGenerator(link.target)}
                                target={targetStringGenerator(link.target)}
                                onClick={() => {
                                  triggerGA(link.category, link.label);
                                }}
                              >
                                {link.span ? (
                                  <>
                                    {link.title}
                                    <span>{link.span}</span>
                                  </>
                                ) : (
                                  <>{link.title}</>
                                )}
                              </a>
                            </li>
                          )))
                          || ''}
                      </ColumnWrapper>
                    </nav>
                  ))}
                </div>
                <div className="col-6 col-md-2 order-2 order-md-3">
                  {/* Company - stacked top */}
                  {columns.slice(1, 2).map((item) => (
                    <div key={uuidv4()}>
                      <nav
                        aria-labelledby={item.arialabelledby}
                        className="mb-5"
                      >
                        <h2 className="footer-col-title" id={item.arialabelledby}>
                          {item.title}
                        </h2>
                        <ColumnWrapper className="column">
                          {(item.items
                            && item.items.map((link) => (
                              <li className="column-row" key={uuidv4()}>
                                <a
                                  className="column-link"
                                  id={link.id}
                                  href={link.url}
                                  rel={relStringGenerator(link.target)}
                                  target={targetStringGenerator(link.target)}
                                  onClick={() => {
                                    triggerGA(link.category, link.label);
                                  }}
                                >
                                  {link.span ? (
                                    <>
                                      {link.title}
                                      <span>{link.span}</span>
                                    </>
                                  ) : (
                                    <>{link.title}</>
                                  )}
                                </a>
                              </li>
                            )))
                            || ''}
                        </ColumnWrapper>
                      </nav>
                    </div>
                  ))}
                  {/* Company stacked bottom */}
                  {columns.slice(2, 3).map((item) => (
                    <div key={uuidv4()}>
                      <nav
                        aria-labelledby={item.arialabelledby}
                        className="mb-5"
                      >
                        <h2 className="footer-col-title" id={item.arialabelledby}>
                          {item.title}
                        </h2>
                        <ColumnWrapper className="column">
                          {(item.items
                            && item.items.map((link) => (
                              <li className="column-row" key={uuidv4()}>
                                <a
                                  className="column-link"
                                  id={link.id}
                                  href={link.url}
                                  rel={relStringGenerator(link.target)}
                                  target={targetStringGenerator(link.target)}
                                  onClick={() => {
                                    triggerGA(link.category, link.label);
                                  }}
                                >
                                  {link.span ? (
                                    <>
                                      {link.title}
                                      <span>{link.span}</span>
                                    </>
                                  ) : (
                                    <>{link.title}</>
                                  )}
                                </a>
                              </li>
                            )))
                            || ''}
                        </ColumnWrapper>
                      </nav>
                    </div>
                  ))}
                </div>
                {/* API Categories */}
                <div className="col-6 col-md-2 order-3 order-md-4">
                  {columns.slice(3, 4).map((item) => (
                    <nav aria-labelledby={item.arialabelledby} key={uuidv4()}>
                      <h2 className="footer-col-title" id={item.arialabelledby}>
                        {item.title}
                      </h2>
                      <ColumnWrapper className="column">
                        {(item.items
                          && item.items.map((link) => (
                            <li className="column-row" key={uuidv4()}>
                              <a
                                className="column-link"
                                id={link.id}
                                href={link.url}
                                rel={relStringGenerator(link.target)}
                                target={targetStringGenerator(link.target)}
                                onClick={() => {
                                  triggerGA(link.category, link.label);
                                }}
                              >
                                {link.span ? (
                                  <>
                                    {link.title}
                                    <span>{link.span}</span>
                                  </>
                                ) : (
                                  <>{link.title}</>
                                )}
                              </a>
                            </li>
                          )))
                          || ''}
                      </ColumnWrapper>
                    </nav>
                  ))}

                </div>
                {/* Social media icons */}
                <div className="col-6 col-md-2 order-4 order-md-5">
                  <div className="row">
                    {columns.slice(4, 5).map((item) => (
                      <div className="col-sm-12" key={uuidv4()}>
                        <nav
                          aria-labelledby={item.arialabelledby}
                          className="mb-5"
                        >
                          <h2 className="footer-col-title" id={item.arialabelledby}>
                            {item.title}
                          </h2>
                          <ColumnWrapper className="column">
                            {(item.items
                              && item.items.map((link) => (
                                <li className="column-row" key={uuidv4()}>
                                  <a
                                    className="column-link"
                                    id={link.id}
                                    href={link.url}
                                    rel={relStringGenerator(link.target)}
                                    target={targetStringGenerator(
                                      link.target,
                                    )}
                                    onClick={() => {
                                      triggerGA(link.category, link.label);
                                    }}
                                  >
                                    <div direction="row" wrap="nowrap">
                                      <SocialSVGWrapper
                                        className="d-inline-block align-self-center social-svg"
                                        // eslint-disable-next-line react/no-danger
                                        dangerouslySetInnerHTML={{
                                          __html: link.svg,
                                        }}
                                      />
                                      {link.span ? (
                                        <>
                                          {link.title}
                                          <span>{link.span}</span>
                                        </>
                                      ) : (
                                        <>
                                          <div className="d-inline-block align-self-center">
                                            {link.title}
                                          </div>
                                        </>
                                      )}
                                    </div>
                                  </a>
                                </li>
                              )))
                              || ''}
                          </ColumnWrapper>
                        </nav>
                      </div>
                    ))}
                  </div>
                </div>
                <div className="col-12 d-block d-md-none text-center order-last">
                  <span className="col-12 copyright">{data.copyright}</span>
                </div>
              </div>
            </div>
          </div>
          {/*  eslint-enable */}
        </div>
      </section>
    </FooterWrapper>
  );
};

export default Footer;
