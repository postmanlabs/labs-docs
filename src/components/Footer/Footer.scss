@import '../Shared/_variables.scss';

.section {
  padding-top: 80px;
  background-color: #FFF;
  // font-size: 1.4rem;
  font-size: 14px;
  // border-top: 1px solid #e6e6e6;
}

footer {
  border-top: 1px solid #e6e6e6;
}

.footer-img {
  @media (min-width: 768px) {
    filter: grayscale(100%);
    transition: all 0.2s ease-in-out 0s; 
    &:hover {
      filter: grayscale(0);
      transition: all 0.2s ease-in-out;
    }
  }
}
.copyright {
  color: #707070;
  // font-size: 1.2rem;
  font-size: 12px
}

.footer-col-title {
  font-size: 16px !important;
  font-family: $Inter600;
  font-weight: 600;
  line-height: 1.4;
  margin-bottom: 8px;
  font-family: $Inter600 !important;
  color: #707070;
}
.column {
  // padding-left: 0;
  margin-left: 0px;
}

.column-row {
  list-style-type: none;
  margin-bottom: 0;
}
.column-link {
  color: #707070;
  text-transform: none;
  font-family: $Inter400;
  font-weight: 400;
  display: block;
  line-height: 1.7;
  padding: 5px 0;
  &:hover {
    color: #0265D2;
    text-decoration: none;
  }

  span {
    border: 1px solid rgb(218, 233, 241);
    border-radius: 30px;
    padding: 0 7px 0 4px;
    display: inline-block;
    white-space: pre;
    margin-left: 6px;
  }
}

.social-svg {
  display: inline-block;
  margin-right: 8px;
  width: 16px;
  height: 16px;

  & svg {
    max-height: 100%;
    width: auto;
  }
}
// .footer {
//   background-color: $deep-space-blue;
//   color: white;
//   list-style-type: none;
//   padding: 32px 8px;
//   @media screen and (min-width: 500px) {
//     padding: 80px 80px;
//   }


//   .footer-top {
//     @media screen and (min-width: 765px) {
//       padding-left: 50px;
//       padding-right: 50px;
//     }
//   }

//   .footer-column {

//     a.span {
//       color:#ffffff !important;
//       font-weight: 700;
//       font-style: italic;
//       &:hover {
//         text-decoration: underline;
//       }
//     }
//     ul {
//       margin-left: 0px !important;
//     }
   
//     .footer-column__title {
//       font-size: 16px !important;
//       margin-bottom: 16px;
//       font-family: 'Open Sans', 'Segoe UI', Tahoma, sans-serif;
//       text-transform: uppercase;
//     }

//     .footer-column__link-wrapper {
//       margin-bottom: 6px !important;
//       margin-top: 2px !important;
//     }
//     .footer-column__list {
//       list-style: none;
//       font-size: 14px;

//       .footer-column__link {
//         color: #ffffff;
//         font-family: 'Open Sans', 'Segoe UI', Tahoma, sans-serif;
//         text-transform: capitalize;
//         padding: 4px 0px 2px !important;

//         &:hover {
//           text-decoration: underline !important;
//           color: #ffffff;
//         }
//       }
//     }
//   }

//   .footer-column__link{
//     font-size: 14px; 
//   }

//   .footer-column__link:hover {
//     color: #ffffff;
//     text-decoration: underline;
//   } 

// .items_border {
//   a { color: #ffffff;
//   }
// }
// .items_line {
//   border-right: 1px solid #ffffff;
//   a:hover {
//     text-decoration: underline;
//   }
// }

// .items_noline {
//   border-right: 1px solid #ffffff;
// }

// .items_noline:last-child {
//   border-right: none;
// }

// @media screen and (max-width: 575px) {
//   .items_line,
//   .items_noline {
//     border-right: none;
//   }
// }

//   // center in mobile view
//   #footer__social, 
//   #footer__copyright {
//     text-align: center;
//   }

//   #footer__social {
//     ul {
//       margin-left: 0px;
//     }
//     @media (min-width: 992px) {
//       text-align: left;
//     }
//   }

//   #footer__copyright {
//     li {
//       margin-right: 4px;
//     }
//     @media (min-width: 992px) {
//       text-align: right;
//     }
//   }

//   .footer__items {
//     border-right: 1px solid #ffffff;
//     padding: 0px 10px;
//   }
//   .footer__items:last-child {
//     border-right: none;
//   }

//   .footer__line {
//     border-top: 1px solid #ffffff;
//     margin: 20px 0px 50px 0px;
//   }
//   .footer__line-wrapper {
//     padding: 0px 50px;
//   }




//   .footer__copyright {
//     font-size: 11px;
//   }

//   .copyright-notice {
//     display: block;
//     padding-right: 16px;

//     @media (min-width: 992px) {
//       display: inline;
//     }
//   }

//   .column__links {
//     list-style-type: none;
//     margin: 0px 0px 0px 0px;; 
//     padding: 4px 0px 2px !important;
//     font-size: 14px;

//     @media (max-width: 576px) {
//       margin: 15px 0;

//       &:first-of-type {
//         margin-top: 0;
//       }
//     }

//     a {
//       color: #ffffff;
//       display: block;
//       line-height: 1.6;
//       padding: 8px 8px 8px 0;
//       text-decoration: none;

//       &:hover {
//         color: $bright-orange;
//         border-bottom: 1px solid;
//       }
//       &.social-icon:hover{
//         border-bottom: none;
//       }

//       /* for social icons */
//       svg {
//         fill: #fff;
//         path {
//           fill: #fff;
//         } 
//         rect {
//           fill: #fff;
//         } 
//         &:hover {
//           path {
//             fill: #F99D78;
//           }
//           rect {
//             fill: #F99D78;
//           } 
//         }
//       }

//       &:hover > .link__hover, 
//       &:hover > .link__hover > .highlight-text {
//         color: #F99D78;
//         padding-bottom: 2px;
//         border-bottom: 1px solid;
//       }
//     }

//     &.column__social {
//       width: 100%;
//       display: flex;
//       flex-direction: row;
//       justify-content: center;
//       padding: 0;

//       @media (min-width: 1200px) {
//         width: auto;
//         padding-left: 20px;
//       }

//       li {
//         margin: 0 10px;

//         &:first-child {
//           margin-left: 0;
//         }

//         &:last-child {
//           margin-right: 0;
//         }
//       }
//     }

//     li {
//       font-size: 14px;
//       line-height: 1.6;
      
//       &.column-item--add-bottom-spacer {
//         margin-bottom: 15px;
//       }
//     }
//   }
// }
