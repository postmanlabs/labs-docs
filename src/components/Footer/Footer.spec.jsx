import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';
import Theme from '../../../styles/theme.jsx'

describe('Footer', () => {
it('renders correctly', () => {
  const tree = renderer
    .create(<Theme><Footer /></Theme>)
    .toJSON();
  expect(tree).toMatchSnapshot();
  });
});
