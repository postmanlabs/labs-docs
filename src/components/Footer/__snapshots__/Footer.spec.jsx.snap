// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="sc-bczRLJ fiqhuZ"
>
  <section
    className="section"
    id="Footer"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-8 offset-sm-2 col-md-12 offset-md-0"
        >
          <div
            className="row"
          >
            <div
              className="sc-dkzDqf ksjPNj col-8 offset-2 col-md-3 offset-md-0 col-lg-2 order-12 order-md-0 pad-md-right align-self-center"
            >
              <img
                alt="Postman"
                className="footer-img mb-5"
                src="https://voyager.postman.com/illustration/postman-footer-rocket-launch.svg"
              />
              <span
                className="col-12 d-none d-md-block copyright"
              >
                Â© 2023 Postman, Inc.
              </span>
            </div>
            <div
              className="col-6 col-md-2 offset-md-1 col-lg-2 offset-lg-2 order-1 order-md-2 mb-5"
            >
              <nav
                aria-labelledby="product"
              >
                <h2
                  className="footer-col-title"
                  id="product"
                >
                  Product
                </h2>
                <ul
                  className="sc-gsnTZi kIEZAF column"
                >
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/product/what-is-postman/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      What is Postman?
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/product/api-repository/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      API repository
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/product/tools/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Tools
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/product/governance/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Governance
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/product/workspaces/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Workspaces
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/product/integrations/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Integrations
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/postman-enterprise/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Enterprise
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/pricing/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Plans and pricing
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/downloads/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Download the app
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/support/"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Support Center
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              className="col-6 col-md-2 order-2 order-md-3"
            >
              <div>
                <nav
                  aria-labelledby="company"
                  className="mb-5"
                >
                  <h2
                    className="footer-col-title"
                    id="company"
                  >
                    Company
                  </h2>
                  <ul
                    className="sc-gsnTZi kIEZAF column"
                  >
                    <li
                      className="column-row"
                    >
                      <a
                        className="column-link"
                        href="https://www.postman.com/company/about-postman/"
                        onClick={[Function]}
                        rel={null}
                        target={null}
                      >
                        About
                      </a>
                    </li>
                    <li
                      className="column-row"
                    >
                      <a
                        className="column-link"
                        href="https://www.postman.com/company/careers/"
                        onClick={[Function]}
                        rel={null}
                        target={null}
                      >
                        Careers and culture
                      </a>
                    </li>
                    <li
                      className="column-row"
                    >
                      <a
                        className="column-link"
                        href="https://www.postman.com/company/press-media/"
                        onClick={[Function]}
                        rel={null}
                        target={null}
                      >
                        Press and media
                      </a>
                    </li>
                    <li
                      className="column-row"
                    >
                      <a
                        className="column-link"
                        href="https://www.postman.com/company/contact-us/"
                        onClick={[Function]}
                        rel={null}
                        target={null}
                      >
                        Contact us
                      </a>
                    </li>
                    <li
                      className="column-row"
                    >
                      <a
                        className="column-link"
                        href="https://www.postman.com/partner-program/"
                        onClick={[Function]}
                        rel={null}
                        target={null}
                      >
                        Partner program
                      </a>
                    </li>
                  </ul>
                </nav>
              </div>
              <div>
                <nav
                  aria-labelledby="security-and-legal"
                  className="mb-5"
                >
                  <h2
                    className="footer-col-title"
                    id="security-and-legal"
                  >
                    Security and Terms
                  </h2>
                  <ul
                    className="sc-gsnTZi kIEZAF column"
                  >
                    <li
                      className="column-row"
                    >
                      <a
                        className="column-link"
                        href="https://www.postman.com/trust/"
                        onClick={[Function]}
                        rel={null}
                        target={null}
                      >
                        Trust and Security
                      </a>
                    </li>
                    <li
                      className="column-row"
                    >
                      <a
                        className="column-link"
                        href="https://www.postman.com/legal/privacy-policy/"
                        onClick={[Function]}
                        rel={null}
                        target={null}
                      >
                        Privacy policy
                      </a>
                    </li>
                    <li
                      className="column-row"
                    >
                      <a
                        className="column-link"
                        href="https://www.postman.com/legal/terms/"
                        onClick={[Function]}
                        rel={null}
                        target={null}
                      >
                        Terms
                      </a>
                    </li>
                  </ul>
                </nav>
              </div>
            </div>
            <div
              className="col-6 col-md-2 order-3 order-md-4"
            >
              <nav
                aria-labelledby="api-categories"
              >
                <h2
                  className="footer-col-title"
                  id="api-categories"
                >
                  API Categories
                </h2>
                <ul
                  className="sc-gsnTZi kIEZAF column"
                >
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/category/payments"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Payments
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/category/financial-services"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Financial Services
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/category/devops"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      DevOps
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/category/developer-productivity"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Developer Productivity
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/category/data-analytics"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Data Analytics
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/category/communication"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Communication
                    </a>
                  </li>
                  <li
                    className="column-row"
                  >
                    <a
                      className="column-link"
                      href="https://www.postman.com/category/artificial-intelligence"
                      onClick={[Function]}
                      rel={null}
                      target={null}
                    >
                      Artificial Intelligence
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              className="col-6 col-md-2 order-4 order-md-5"
            >
              <div
                className="row"
              >
                <div
                  className="col-sm-12"
                >
                  <nav
                    aria-labelledby="social"
                    className="mb-5"
                  >
                    <h2
                      className="footer-col-title"
                      id="social"
                    >
                      Social
                    </h2>
                    <ul
                      className="sc-gsnTZi kIEZAF column"
                    >
                      <li
                        className="column-row"
                      >
                        <a
                          className="column-link"
                          href="https://twitter.com/getpostman"
                          onClick={[Function]}
                          rel={null}
                          target={null}
                        >
                          <div
                            direction="row"
                            wrap="nowrap"
                          >
                            <div
                              className="sc-hKMtZM dZiXNm d-inline-block align-self-center social-svg"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "<svg width='16' height='14' viewBox='0 0 16 14' fill='none' xmlns='http://www.w3.org/2000/svg'><path d='M14.1439 3.76852C14.1534 3.9068 14.1534 4.04509 14.1534 4.18465C14.1534 8.43705 10.9162 13.3414 4.99671 13.3414V13.3388C3.24808 13.3414 1.53578 12.8405 0.0637207 11.8961C0.317985 11.9267 0.573525 11.9419 0.829701 11.9426C2.27882 11.9439 3.68651 11.4576 4.82656 10.5623C3.44946 10.5362 2.24186 9.63827 1.82 8.32744C2.3024 8.42048 2.79946 8.40136 3.27294 8.272C1.77156 7.96867 0.691417 6.64955 0.691417 5.11759C0.691417 5.10357 0.691417 5.09018 0.691417 5.0768C1.13877 5.32597 1.63965 5.46425 2.15201 5.47955C0.737936 4.5345 0.302054 2.65332 1.15598 1.18254C2.7899 3.19308 5.20063 4.41533 7.78853 4.54469C7.52916 3.42695 7.88348 2.25568 8.71955 1.46994C10.0157 0.251509 12.0543 0.31396 13.2727 1.6095C13.9935 1.46739 14.6843 1.20293 15.3164 0.828224C15.0762 1.57318 14.5734 2.20597 13.9017 2.60808C14.5396 2.53288 15.1628 2.3621 15.7498 2.10146C15.3177 2.74891 14.7735 3.31288 14.1439 3.76852Z' fill='#1D9BF0'/></svg>",
                                }
                              }
                            />
                            <div
                              className="d-inline-block align-self-center"
                            >
                              Twitter
                            </div>
                          </div>
                        </a>
                      </li>
                      <li
                        className="column-row"
                      >
                        <a
                          className="column-link"
                          href="https://www.linkedin.com/company/postman-platform"
                          onClick={[Function]}
                          rel={null}
                          target={null}
                        >
                          <div
                            direction="row"
                            wrap="nowrap"
                          >
                            <div
                              className="sc-hKMtZM dZiXNm d-inline-block align-self-center social-svg"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "<svg width=\\"25\\" height=\\"25\\" viewBox=\\"0 0 25 25\\" xmlns=\\"http://www.w3.org/2000/svg\\"><path d=\\"M5.134 7.363H.537v14.044h4.597zM5.437 3.019C5.407 1.642 4.44.593 2.866.593 1.293.593.265 1.642.265 3.019c0 1.348.998 2.427 2.541 2.427h.03c1.603 0 2.601-1.079 2.601-2.427zM21.714 13.354c0-4.313-2.268-6.32-5.293-6.32-2.44 0-3.534 1.362-4.144 2.319v-1.99H7.68c.06 1.318 0 14.044 0 14.044h4.598v-7.843c0-.42.03-.839.151-1.139.333-.839 1.09-1.707 2.36-1.707 1.664 0 2.329 1.288 2.329 3.175v7.514h4.597v-8.053z\\" fill=\\"#0e76a8\\" fillRule=\\"evenodd\\"></path></svg>",
                                }
                              }
                            />
                            <div
                              className="d-inline-block align-self-center"
                            >
                              LinkedIn
                            </div>
                          </div>
                        </a>
                      </li>
                      <li
                        className="column-row"
                      >
                        <a
                          className="column-link"
                          href="https://github.com/postmanlabs"
                          onClick={[Function]}
                          rel={null}
                          target={null}
                        >
                          <div
                            direction="row"
                            wrap="nowrap"
                          >
                            <div
                              className="sc-hKMtZM dZiXNm d-inline-block align-self-center social-svg"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "<svg height=\\"16\\" class=\\"octicon octicon-mark-github\\" viewBox=\\"0 0 16 16\\" version=\\"1.1\\" width=\\"16\\" aria-hidden=\\"true\\"><path fill=\\"#24292f\\" fillRule=\\"evenodd\\" d=\\"M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z\\"></path></svg>",
                                }
                              }
                            />
                            <div
                              className="d-inline-block align-self-center"
                            >
                              GitHub
                            </div>
                          </div>
                        </a>
                      </li>
                      <li
                        className="column-row"
                      >
                        <a
                          className="column-link"
                          href="https://www.youtube.com/c/Postman"
                          onClick={[Function]}
                          rel={null}
                          target={null}
                        >
                          <div
                            direction="row"
                            wrap="nowrap"
                          >
                            <div
                              className="sc-hKMtZM dZiXNm d-inline-block align-self-center social-svg"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "<svg height='16' width='16' enable-background='new 0 0 30 21.14' viewBox='0 0 30 21.14' xmlns='http://www.w3.org/2000/svg'><path d='m29.37 3.3c-.35-1.3-1.36-2.32-2.65-2.67-2.34-.63-11.72-.63-11.72-.63s-9.38 0-11.72.63c-1.29.35-2.31 1.37-2.65 2.67-.63 2.36-.63 7.27-.63 7.27s0 4.91.63 7.27c.35 1.3 1.36 2.32 2.65 2.67 2.34.63 11.72.63 11.72.63s9.38 0 11.72-.63c1.29-.35 2.31-1.37 2.65-2.67.63-2.36.63-7.27.63-7.27s0-4.91-.63-7.27zm-17.44 11.73v-8.92l7.84 4.46z' fill='#c4302b'></path></svg>",
                                }
                              }
                            />
                            <div
                              className="d-inline-block align-self-center"
                            >
                              YouTube
                            </div>
                          </div>
                        </a>
                      </li>
                      <li
                        className="column-row"
                      >
                        <a
                          className="column-link"
                          href="https://www.twitch.tv/getpostman"
                          onClick={[Function]}
                          rel={null}
                          target={null}
                        >
                          <div
                            direction="row"
                            wrap="nowrap"
                          >
                            <div
                              className="sc-hKMtZM dZiXNm d-inline-block align-self-center social-svg"
                              dangerouslySetInnerHTML={
                                Object {
                                  "__html": "<svg version='1.1' id='Layer_1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' x='0px' y='0px' viewBox='0 0 2400 2800' style='enable-background:new 0 0 2400 2800;' xml:space='preserve'><style type='text/css'>.st0{fill:#FFFFFF;}.st1{fill:#9146FF;}</style><title>Asset 2</title><g><polygon class='st0' points='2200,1300 1800,1700 1400,1700 1050,2050 1050,1700 600,1700 600,200 2200,200'/><g><g id='Layer_1-2'><path class='st1' d='M500,0L0,500v1800h600v500l500-500h400l900-900V0H500z M2200,1300l-400,400h-400l-350,350v-350H600V200h1600 V1300z'/><rect x='1700' y='550' class='st1' width='200' height='600'/><rect x='1150' y='550' class='st1' width='200' height='600'/></g></g></g></svg>",
                                }
                              }
                            />
                            <div
                              className="d-inline-block align-self-center"
                            >
                              Twitch
                            </div>
                          </div>
                        </a>
                      </li>
                    </ul>
                  </nav>
                </div>
              </div>
            </div>
            <div
              className="col-12 d-block d-md-none text-center order-last"
            >
              <span
                className="col-12 copyright"
              >
                Â© 2023 Postman, Inc.
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</footer>
`;
